app/src/main/java/org/kinecosystem/kinit/view/MainActivity.java

```diff
@@ -42,27 +54,74 @@ public static Intent getIntent(Context context) {
     protected void onCreate(Bundle savedInstanceState) {
         KinitApplication.coreComponent.inject(this);
         super.onCreate(savedInstanceState);
-        overridePendingTransition(R.anim.fade_in, R.anim.fade_out);
+        int selectedTabIndex = (savedInstanceState != null) ? savedInstanceState.getInt(SELECTED_TAB_INDEX_KEY, 0) : 0;
         setContentView(R.layout.main_activity);
         bottomTabNavigation = findViewById(R.id.navigation);
-        int selectedTabIndex = (savedInstanceState != null) ? savedInstanceState.getInt(SELECTED_TAB_INDEX_KEY, 0) : 0;
         bottomTabNavigation.setSelectedTabIndex(selectedTabIndex);
-        enablePageSelection();
-        tabsAdapter = new TabsAdapter();
-        tabsAdapter.setPageSelectionListener(this);
         viewPager = findViewById(R.id.view_pager);
+        tabsAdapter = new TabsAdapter();
         viewPager.setAdapter(tabsAdapter);
-        if (!tabsAdapter.shouldShowAnimation()) {
-            viewPager.setCurrentItem(TabsAdapter.EARN_TAB_INDEX, false);
+        enablePageSelection();
+        if (shouldShowPreEarnAnimation()) {
+            disablePageSelection();
+            showPreEarnAnimation();
+        } else {
+            readyToEarnText = findViewById(R.id.ready_to_earn_image);
+            ((ViewGroup) readyToEarnText.getParent()).removeView(readyToEarnText);
         }
+        overridePendingTransition(R.anim.fade_in, R.anim.fade_out);
         handleIntentExtras(getIntent().getExtras());
     }
 
+    private boolean shouldShowPreEarnAnimation() {
+        return taskRepository.isTaskAvailable() && !taskRepository.isTaskStarted().get();
+    }
+
+    private void showPreEarnAnimation() {
+        viewPager.setVisibility(View.INVISIBLE);
+        Animation slideLeftIn = AnimationUtils.loadAnimation(this, R.anim.slide_left_in);
+        Animation slideLeftOut = AnimationUtils.loadAnimation(this, R.anim.slide_left_out);
+
+        slideLeftIn.setDuration(PRE_ANIMATION_DURATION);
+        slideLeftOut.setDuration(PRE_ANIMATION_DURATION);
+
+        slideLeftIn.setInterpolator(new AccelerateDecelerateInterpolator());
+        slideLeftOut.setInterpolator(slideLeftIn.getInterpolator());
+
+        slideLeftIn.setAnimationListener(new Animation.AnimationListener() {
+            @Override
+            public void onAnimationStart(Animation animation) {
+                enablePageSelection();
+            }
+
+            @Override
+            public void onAnimationEnd(Animation animation) {
+                enablePageSelection();
+            }
+
+            @Override
+            public void onAnimationRepeat(Animation animation) {
+
+            }
+        });
+
+        viewPager.postDelayed(() -> {
+            viewPager.setVisibility(View.VISIBLE);
```

Comment:
`viewPager` can be null if `postDelay` have been called and the user pressed back immediately. Maybe will be better to wrap with null check.